testutils/src/main/scala/za/co/absa/RandomDataGenerator.scala

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa
+
+import java.io._
+import org.joda.time.DateTime
+import org.joda.time.format.{DateTimeFormat, DateTimeFormatter}
+import scala.util.Random
+import Numeric.Implicits._
+
+/* Generate data for testing
+ * Data contains a given number of columns and rows
+ * Coulmns are of type: string, int, date, float, boolean
+ * First 6 columns have predefined names and types, the rest are repeated using a pattern
+ * Data are generated at random or using simple rules (eg x = row * 100000 + col})
+ */
+
+object RandomDataGenerator {
+
+  // Set DateTime format to be used
+  val dtf: DateTimeFormatter = DateTimeFormat.forPattern("yyyy-MM-dd'T'HH:mm:ss")
+
+  // Predefined columns
+  val reservedColumns: Map[Int,String] = Map(
+    1 -> "key",
+    2 -> "id",
+    3 -> "version",
+    4 -> "Date",
+    5 -> "value",
+    6 -> "address")
+
+  // Initialize random generator
+  private val random = new Random(System.nanoTime)
```

Comment:
I don't think `new Random()` needs a seed like `System.nanoTime` since it is pretty good at supplying it's own random seed, see (or similar depending on JDK version):
 [`public Random() { this(System.currentTimeMillis()); }`](http://developer.classpath.org/doc/java/util/Random-source.html)